src/common/transformations/src/transformations/common_optimizations/fuse_rotary_positional_embeddings.cpp

```diff
@@ -427,6 +428,79 @@ ov::pass::RoPEFusionPreprocess::RoPEFusionPreprocess() {
     this->register_matcher(m, callback);
 }
 
+ov::pass::RoPEFusionVIT3D::RoPEFusionVIT3D() {
+    using namespace ov::op::util;
+    MATCHER_SCOPE(RoPEFusionVIT3D);
+
+    auto x = pattern::any_input(pattern::rank_equals(3));
+    auto x_or_cos1 = pattern::any_input(pattern::rank_equals(3));
+    auto x_or_cos2 = pattern::any_input(pattern::rank_equals(3));
+    auto t_sin = pattern::any_input(pattern::rank_equals(3));
+
+    auto varsplit = pattern::wrap_type<v1::VariadicSplit>({x, 2, {"half_ndims", "?"}});
+    varsplit->set_output_size(2);
+
+    auto int32_max = std::numeric_limits<std::int32_t>::max();
+
+    auto x2 = NewGenSlice(x, "half_ndims", int32_max, 1, 2);
+    auto x2neg = pattern::wrap_type<v1::Multiply>({x2 | varsplit->output(1), -1.0f}, {{"auto_broadcast", "numpy"}});
+    auto x1 = NewGenSlice(x, 0, "half_ndims", 1, 2);
+    auto x_rotate_half = pattern::wrap_type<v0::Concat>({x2neg, x1 | varsplit->output(0)}, {{"axis", -1}});
+
+    auto mul_cos = pattern::wrap_type<v1::Multiply>({x_or_cos1, x_or_cos2}, {{"auto_broadcast", "numpy"}});
```

Comment:
For `mul_cos`, if you want to switch `lefthand` and `righthand` of the `Multiply` maybe a `or` is better
```c++
auto mul_cos1 = pattern::wrap_type<v1::Multiply>({x, cos}, {{"auto_broadcast", "numpy"}});
auto mul_cos2 = pattern::wrap_type<v1::Multiply>({cos, x}, {{"auto_broadcast", "numpy"}});
auto mul_cos = mul_cos1 | mul_cos2 
```